Assets/Project/Modules/Enemies/Slime/Scripts/SlimeRefactor/SlimeAnimation.cs

```diff
@@ -0,0 +1,69 @@
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+public class SlimeAnimation : ISlimeComponent
+{
+    
+    [SerializeField] private GameObject _explosionParticles;
+    
+    [SerializeField] private float _squashAmount = 0.8f;
+    [SerializeField] private float _stretchAmount = 1.2f;
+    [SerializeField] private float _animationSpeed = 5f;
+
+    private bool _doSquashing = true;
+    private bool _isSquashing = false;
+
+
+    private void Update()
+    {
+        if (_doSquashing)  
+        {
+            if (!_isSquashing)
+            {
+                StartCoroutine(SquashAndStretch());
+            }
+        }
+    }
```

Comment:
This isn't very clean.
With the functions I described above this automatic checking can be removed.